### ISSUE TITLE
More detailed requirements

### ISSUE THREAD

**Issue by rooknpown** [kaist-dmlab/Prune4Rel#10]
Installing the requirements.txt is not enough. Could you provide docker environment or more detailed requirements?

**Comment by seolachoi** [comment]
We conducted our experiments using conda environments, and we’ve added additional setups for debugging and other implementations within that environment. Therefore, providing a specific set of requirements solely for this code is difficult.

However, if the existing requirements.txt doesn’t solve the issue, you can refer to the additional requirements below. Please note that this environment includes some unnecessary packages, so we recommend only installing the packages required by the provided requirements.txt.

Let me know if you have any further questions or need more help!

accelerate==0.28.0
aepo==0.1.6
aioboto3==12.1.0
aiobotocore==2.8.0
aiohttp==3.8.5
aiohttp-retry==2.8.3
aioitertools==0.11.0
aiosignal==1.3.1
alembic==1.12.0
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==4.0.0
appdirs==1.4.4
asttokens==2.4.1
async-timeout==4.0.3
attrs==23.1.0
banal==1.0.6
beartype==0.17.0
beautifulsoup4==4.12.2
boto3==1.33.1
botocore==1.33.1
cachetools==5.3.1
certifi==2023.7.22
cffi==1.15.1
charset-normalizer==3.2.0
click==8.1.7
cmake==3.27.5
comm==0.2.2
config==0.5.1
contourpy==1.2.0
cryptography==41.0.4
cssselect==1.2.0
cssutils==2.7.1
cycler==0.12.1
dataclasses-json==0.6.7
dataset==1.6.2
datasets==2.12.0
debugpy==1.8.1
decorator==5.1.1
deeplake==3.8.14
dill==0.3.6
distro==1.9.0
docker-pycreds==0.4.0
ema-pytorch==0.3.3
evaluate==0.4.2
exceptiongroup==1.1.3
executing==2.0.1
faiss-gpu==1.7.2
filelock==3.12.4
fonttools==4.47.0
frozenlist==1.4.0
fsspec==2023.6.0
gitdb==4.0.11
GitPython==3.1.42
greenlet==2.0.2
h11==0.14.0
httpcore==0.17.3
httpx==0.24.1
huggingface-hub==0.23.4
humbug==0.3.2
hydra-core==1.3.2
idna==3.4
importlib-metadata==6.8.0
importlib-resources==6.1.1
ipykernel==6.23.1
ipython==8.18.1
jaraco.classes==3.3.0
jedi==0.19.1
jeepney==0.8.0
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.3.2
jupyter_client==8.6.1
jupyter_core==5.7.2
keyring==24.2.0
kiwisolver==1.4.5
knockknock==0.1.8.1
libdeeplake==0.0.95
lit==16.0.6
llm-blender==0.0.2
lxml==4.9.3
lz4==4.3.3
Mako==1.2.4
MarkupSafe==2.1.3
marshmallow==3.21.3
matplotlib==3.8.2
matplotlib-inline==0.1.6
matrix-client==0.4.0
more-itertools==10.1.0
mpmath==1.3.0
multidict==6.0.4
multiprocess==0.70.14
mypy-extensions==1.0.0
nest-asyncio==1.5.8
networkx==3.1
numpy==1.26.4
nvidia-cublas-cu11==11.10.3.66
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu11==8.5.0.96
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu11==10.9.0.58
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu11==10.2.10.91
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu11==11.7.4.91
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu11==2.14.3
nvidia-nccl-cu12==2.20.5
nvidia-nvjitlink-cu12==12.5.40
nvidia-nvtx-cu11==11.7.91
nvidia-nvtx-cu12==12.1.105
omegaconf==2.3.0
openai==0.28.0
packaging==23.1
pandas==2.1.1
parso==0.8.3
pathos==0.3.0
pathtools==0.1.2
peft==0.9.0
pexpect==4.9.0
Pillow==10.0.1
pip==23.2.1
platformdirs==4.2.0
pox==0.3.3
ppft==1.7.6.7
premailer==3.10.0
prompt-toolkit==3.0.43
protobuf==4.25.3
psutil==5.9.8
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==13.0.0
pycparser==2.21
pydantic==2.5.3
pydantic_core==2.14.6
Pygments==2.17.2
PyJWT==2.8.0
pyparsing==3.1.1
PyQt5==5.15.10
PyQt5-Qt5==5.15.2
PyQt5-sip==12.13.0
python-dateutil==2.8.2
python-telegram-bot==20.5
pytorch-pretrained-vit==0.0.7
pytz==2023.3.post1
PyYAML==6.0.1
pyzmq==25.1.2
randaugment==1.0.2
regex==2023.8.8
requests==2.31.0
responses==0.18.0
s3transfer==0.8.0
safetensors==0.4.3
scikit-learn==1.4.1.post1
scipy==1.11.3
seaborn==0.13.0
SecretStorage==3.3.3
sentence-transformers==3.0.1
sentencepiece==0.2.0
sentry-sdk==1.41.0
setproctitle==1.3.3
setuptools==68.0.0
six==1.16.0
smmap==5.0.1
sniffio==1.3.0
soupsieve==2.5
split-folders==0.5.1
SQLAlchemy==1.4.49
stack-data==0.6.3
sympy==1.12
tensor-parallel==1.2.4
threadpoolctl==3.3.0
tokenizers==0.19.1
torch==2.3.1
torchfile==0.1.0
torchvision==0.15.2
tornado==6.4
tqdm==4.66.4
traitlets==5.14.2
transformers==4.41.2
triton==2.3.1
twilio==8.8.0
typing_extensions==4.8.0
typing-inspect==0.9.0
tzdata==2023.3
urllib3==1.26.16
wandb==0.15.3
wcwidth==0.2.13
wheel==0.38.4
wrapt==1.16.0
xxhash==3.3.0
yagmail==0.15.293
yarl==1.9.2
zipp==3.17.0
